src/Azure/Orleans.Persistence.AzureStorage/Providers/Storage/AzureBlobStorageOptions.cs

```diff
@@ -1,35 +1,80 @@
 using System;
+using System.Threading.Tasks;
+using Azure;
 using Azure.Core;
-using Microsoft.Azure.Cosmos.Table;
+using Azure.Storage;
+using Azure.Storage.Blobs;
 using Newtonsoft.Json;
 using Orleans.Persistence.AzureStorage;
 using Orleans.Runtime;
 
 namespace Orleans.Configuration
 {
-    public class AzureBlobStorageOptions
+    public class AzureBlobStorageOptions : AzureBlobUtils.IBlobServiceClientOptions
```

Comment:
This class (like the one for Table) is starting to be really hard to understand... what about just exposing a factory method, with some helpers function to use a connection string, or a token credentials, ect. ?

Reply:
Yes, I agree. I'll add helper methods for each valid configuration to configure it instead, maybe called "SetCredentials"?

Reply:
I've added helpers in the latest commit. Let me know what you think